### ISSUE TITLE
repeated disconnections when used with ZenPad 3S 10

### ISSUE THREAD

**Issue by fchyzak** [yangh/sevensquare#16]
Hi, I've just tried SevenSquare (commit 40da13dd8a0c5134c619a645d199e06e48ddfd90), compiled on an archlinux (4.13.12-1-ARCH), against an ASUS ZenPad 3S 10 (Z500M, a.k.a. P027). It compiles well, it does a capture of my tablet's screen, but with a constant overlay, cycling with "Waiting...", "Probing device...", "Connected...", etc. At the same time, I get the following warnings on the console (again, a sort of a cycle). (Disconnections occur every second or so.)

Is there anything I should try to fix this?

<pre>
1511855561218 ADB Wait for device 
1511855561222 ADB Found 
1511855561224 Compressed data transfer false 
New view size QSize(320, 562) QSize(-1, -1) 
1511855561261 New scene size: 320 530 
1511855561354 Screencap on device options -q -s false false 
1511855561878 CAP FB in 524 ms 
1511855561901 New Remote screen FB: 1536 2048 1 
New view size QSize(320, 458) QSize(320, 562) 
1511855561917 New scene size: 319 426 
1511855561998 Adb exec error... 
1511855561998 Device disconnected 
1511855562030 Device has sys LCD brightness API: true 
1511855562486 CAP FB in 562 ms 
1511855562619 ADB Wait for device 
1511855562624 ADB Found 
1511855562627 Compressed data transfer false 
1511855562788 Screencap on device options -q -s false false 
1511855563340 CAP FB in 551 ms 
1511855563457 Adb exec error... 
1511855563457 Device disconnected 
1511855563490 Device has sys LCD brightness API: true 
1511855563916 CAP FB in 540 ms 
1511855564395 ADB Wait for device 
1511855564405 ADB Found 
1511855564408 Compressed data transfer false 
1511855564537 Screencap on device options -q -s false false 
1511855565059 CAP FB in 522 ms 
1511855565166 Adb exec error... 
1511855565166 Device disconnected 
1511855565200 Device has sys LCD brightness API: true 
1511855565693 CAP FB in 599 ms 
1511855566074 ADB Wait for device 
1511855566081 ADB Found 
</pre>

**Comment by yangh** [comment]
it seem that adb has some error.

On Tuesday, November 28, 2017, fchyzak <notifications@github.com> wrote:
> Hi, I've just tried SevenSquare (commit 40da13d), compiled on an
archlinux (4.13.12-1-ARCH), against an ASUS ZenPad 3S 10 (Z500M, a.k.a.
P027). It compiles well, it does a capture of my tablet's screen, but with
a constant overlay, cycling with "Waiting...", "Probing device...",
"Connected...", etc. At the same time, I get the following warnings on the
console (again, a sort of a cycle).
>
> Is there anything I should try to fix this?
>
> 1511855561218 ADB Wait for device
> 1511855561222 ADB Found
> 1511855561224 Compressed data transfer false
> New view size QSize(320, 562) QSize(-1, -1)
> 1511855561261 New scene size: 320 530
> 1511855561354 Screencap on device options -q -s false false
> 1511855561878 CAP FB in 524 ms
> 1511855561901 New Remote screen FB: 1536 2048 1
> New view size QSize(320, 458) QSize(320, 562)
> 1511855561917 New scene size: 319 426
> 1511855561998 Adb exec error...
> 1511855561998 Device disconnected
> 1511855562030 Device has sys LCD brightness API: true
> 1511855562486 CAP FB in 562 ms
> 1511855562619 ADB Wait for device
> 1511855562624 ADB Found
> 1511855562627 Compressed data transfer false
> 1511855562788 Screencap on device options -q -s false false
> 1511855563340 CAP FB in 551 ms
> 1511855563457 Adb exec error...
> 1511855563457 Device disconnected
> 1511855563490 Device has sys LCD brightness API: true
> 1511855563916 CAP FB in 540 ms
> 1511855564395 ADB Wait for device
> 1511855564405 ADB Found
> 1511855564408 Compressed data transfer false
> 1511855564537 Screencap on device options -q -s false false
> 1511855565059 CAP FB in 522 ms
> 1511855565166 Adb exec error...
> 1511855565166 Device disconnected
> 1511855565200 Device has sys LCD brightness API: true
> 1511855565693 CAP FB in 599 ms
> 1511855566074 ADB Wait for device
> 1511855566081 ADB Found
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.<
https://ci4.googleusercontent.com/proxy/PvYIWi3dkUXzVvtswGpaQrDz3r1JMgaIoLn5UfvIM9_rHdJiSU-bDf7N-kYpumV_Y7ZmHHCOw3zNbESMBEWwMosCUS5TEkmDIsWd-yOnVVuKnp_cszBe-reKM5Z-_qk5Qt7B-GSoXduwS_2GAFJnz6Y1o1a3Ww=s0-d-e1-ft#https://github.com/notifications/beacon/AAC7_rJqHMvWxXC0cMbTFkEboD61Lq7jks5s676tgaJpZM4Qs1Px.gif>

**Comment by ohthehugemanatee** [comment]
Same issue here on Ubuntu 18.04 and Samsung S8. 

```
Gtk-Message: Failed to load module "overlay-scrollbar"
1531928542193 ADB Wait for device 
1531928542196 ADB Found 
1531928542198 Compressed data transfer false 
New view size QSize(320, 562) QSize(-1, -1) 
1531928542235 New scene size: 320 530 
New view size QSize(1036, 1714) QSize(320, 562) 
1531928542252 New scene size: 1015 1682 
1531928542463 Screencap on device options -q -s false false 
1531928543139 CAP FB in 676 ms 
1531928543176 New Remote screen FB: 1080 2220 1 
New view size QSize(1015, 2118) QSize(1036, 1714) 
1531928543215 New scene size: 1014 2086 
1531928543325 Adb exec error... 
1531928543325 Device disconnected 
New view size QSize(1036, 1714) QSize(1015, 2118) 
1531928543327 New scene size: 818 1682 
1531928543899 CAP FB in 653 ms 
1531928544030 ADB Wait for device 
1531928544041 ADB Found 
1531928544049 Compressed data transfer false 
1531928544283 Screencap on device options -q -s false false 
1531928544897 CAP FB in 613 ms 
1531928545046 Adb exec error... 
1531928545047 Device disconnected 
... etc ad nauseum
```
Android Debug Bridge version 1.0.36, Revision 1:7.0.0+r33-2 . Is there a verbose or a debug flag in sevensquare? Any way to get more information on what ADB exec command is tried?

**Comment by hapesurya** [comment]
Same issue on my Ubuntu 18.04.1 LTS and Moto E4 Plus (Android 7.1.1)

~$ seven-square
1548671833895 ADB Wait for device 
New view size QSize(320, 562) QSize(-1, -1) 
1548671833902 New scene size: 320 530 
1548671837119 ADB Found 
1548671837124 Compressed data transfer false 
1548671837636 Screencap on device options -q -s false false 
1548671838138 CAP FB in 502 ms 
1548671838152 New Remote screen FB: 720 1280 1 
New view size QSize(320, 600) QSize(320, 562) 
1548671838168 New scene size: 319 568 
1548671838268 Adb exec error... 
1548671838268 Device disconnected 
1548671838681 CAP FB in 505 ms 
1548671838800 ADB Wait for device 
1548671838811 ADB Found 
1548671838817 Compressed data transfer false 
1548671839141 Screencap on device options -q -s false false 
1548671839645 CAP FB in 504 ms 
1548671839781 Adb exec error... 
1548671839783 Device disconnected 
1548671840173 CAP FB in 512 ms 
1548671840641 ADB Wait for device 
1548671840653 ADB Found 

Need your help please...

**Comment by yangh** [comment]
Fixed in latest commit 1333e50
